### ISSUE TITLE
du show zero size for all files

### ISSUE THREAD

**Issue by nirs** [nirs/slowfs#2]
# du -sh e9cb327f-fa1a-47db-9d1e-27ca648fc20f/*

0   e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df
0   e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df.lease
0   e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df.meta
# ls -lh e9cb327f-fa1a-47db-9d1e-27ca648fc20f/*

-rw-rw----. 77 root root 1.3G Oct 23 05:59 e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df
-rw-rw----. 77 root root 1.0M Oct 23 05:54 e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df.lease
-rw-r--r--. 77 root root  312 Oct 23 05:59 e9cb327f-fa1a-47db-9d1e-27ca648fc20f/8479593f-3dc8-4488-aa2e-4cba822f35df.meta

**Comment by nirs** [comment]
## du in slowfs

```
$ strace du -sh .
execve("/usr/bin/du", ["du", "-sh", "."], [/* 58 vars */]) = 0
...
stat("/tmp/slowfs", {st_mode=S_IFDIR|0775, st_size=100, ...}) = 0
newfstatat(AT_FDCWD, "link_to_target", {st_mode=S_IFREG|0775, st_size=21, ...}, AT_SYMLINK_NOFOLLOW) = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6b36d98000
write(1, "0\tlink_to_target\n", 170 link_to_target
)     = 17
newfstatat(AT_FDCWD, "symlink_to_target", {st_mode=S_IFLNK|0777, st_size=6, ...}, AT_SYMLINK_NOFOLLOW) = 0
write(1, "0\tsymlink_to_target\n", 200  symlink_to_target
)  = 20
newfstatat(AT_FDCWD, "target", {st_mode=S_IFREG|0775, st_size=21, ...}, AT_SYMLINK_NOFOLLOW) = 0
write(1, "0\ttarget\n", 90  target
)              = 9
```

## du in realfs

```
$ strace du -sh *
...
stat("/tmp/slowfs", {st_mode=S_IFDIR|0775, st_size=100, ...}) = 0
newfstatat(AT_FDCWD, "link_to_target", {st_mode=S_IFREG|0775, st_size=21, ...}, AT_SYMLINK_NOFOLLOW) = 0
fstat(1, {st_mode=S_IFCHR|0620, st_rdev=makedev(136, 1), ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f5a4fbb6000
write(1, "4.0K\tlink_to_target\n", 204.0K   link_to_target
)  = 20
newfstatat(AT_FDCWD, "symlink_to_target", {st_mode=S_IFLNK|0777, st_size=6, ...}, AT_SYMLINK_NOFOLLOW) = 0
write(1, "0\tsymlink_to_target\n", 200  symlink_to_target
)  = 20
newfstatat(AT_FDCWD, "target", {st_mode=S_IFREG|0775, st_size=21, ...}, AT_SYMLINK_NOFOLLOW) = 0
```

**Comment by nirs** [comment]
## Log during du call

```
2015-10-24 04:35:50,977 DEBUG [fs] -> opendir u'/' ()
2015-10-24 04:35:50,978 DEBUG [fs] <- opendir 0
2015-10-24 04:35:50,978 DEBUG [fs] -> readdir u'/' (0L,)
2015-10-24 04:35:50,978 DEBUG [fs] <- readdir ['.', '..', u'symlink_to_target', u'link_to_target', u'target']
2015-10-24 04:35:50,979 DEBUG [fs] -> releasedir u'/' (0L,)
2015-10-24 04:35:50,979 DEBUG [fs] <- releasedir 0
2015-10-24 04:35:50,988 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:35:50,988 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650550.977305}
2015-10-24 04:35:50,989 DEBUG [fs] -> getattr u'/link_to_target' (None,)
2015-10-24 04:35:50,989 DEBUG [fs] <- getattr {'st_ctime': 1445644419.0037522, 'st_mtime': 1445644419.0037522, 'st_nlink': 2, 'st_mode': 33277, 'st_size': 21, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445649616.927488}
2015-10-24 04:35:50,990 DEBUG [fs] -> getattr u'/symlink_to_target' (None,)
2015-10-24 04:35:50,990 DEBUG [fs] <- getattr {'st_ctime': 1445638537.539217, 'st_mtime': 1445638296.0826027, 'st_nlink': 1, 'st_mode': 41471, 'st_size': 6, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445649820.8906317}
2015-10-24 04:35:50,990 DEBUG [fs] -> getattr u'/target' (None,)
2015-10-24 04:35:50,991 DEBUG [fs] <- getattr {'st_ctime': 1445644419.0037522, 'st_mtime': 1445644419.0037522, 'st_nlink': 2, 'st_mode': 33277, 'st_size': 21, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445649616.927488}
2015-10-24 04:39:33,548 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:39:33,548 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650550.977305}
2015-10-24 04:39:34,569 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:39:34,570 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650550.977305}
2015-10-24 04:39:43,289 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:39:43,289 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650783.2787461}
2015-10-24 04:42:45,234 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:42:45,234 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650965.2306585}
2015-10-24 04:43:05,258 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:43:05,258 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650981.7258317}
2015-10-24 04:43:10,302 DEBUG [fs] -> getattr u'/' (None,)
2015-10-24 04:43:10,303 DEBUG [fs] <- getattr {'st_ctime': 1445644863.6985278, 'st_mtime': 1445644863.6985278, 'st_nlink': 2, 'st_mode': 16893, 'st_size': 100, 'st_gid': 1000, 'st_uid': 1000, 'st_atime': 1445650990.2989218}
```

**Comment by nirs** [comment]
Fixed in 5d843f049614466046f98008b2badb6a3e2e7580
